### PR TITLE
(QENG-4179) Add HuaweiOS 6 platform

### PR DESCRIPTION
**Pull request by nwolfe**

### CODE REVIEW

**Comment by nwolfe** [comment]
@johnduarte @LuvCurves See https://github.com/puppetlabs/beaker-hostgenerator/pull/46/commits/3773715fabedfd56888213af22b88f90524df409 for changes

@waynr Haha yep....447 files changes.  Like usual, I just blew away the entire `generated/` directory and regenerated everything so we get some single and multihost fixtures with it.

File: lib/beaker-hostgenerator/cli.rb [hunk 1]
```diff
@@ -184,7 +184,7 @@ def supported_values_help_text
       result << "   32    => 32-bit OS\n"
       result << "   64    => 64-bit OS\n"
       result << "   6432  => 64-bit OS with 32-bit Puppet (Windows Only)\n"
-      result << "   POWER => power OS (AIX only)\n"
+      result << "   POWER => power OS (AIX and Huawei only)\n"
```
Comment (johnduarte) [inline lib/beaker-hostgenerator/cli.rb]: Actually, AIX we use power, but for Huawei we use powerpc

File: lib/beaker-hostgenerator/data.rb [hunk 2]
```diff
@@ -339,6 +339,11 @@ def osinfo
             'template' => 'fedora-24-x86_64'
           }
         },
+        'huaweios6-POWER' => {
+          :general => {
+            'platform' => 'huaweios-6-power'
```
Comment (johnduarte) [inline lib/beaker-hostgenerator/data.rb]: I believe this should be `huaweios-6-powerpc`

Comment (nwolfe) [inline lib/beaker-hostgenerator/data.rb]: Oops, good catch, forgot the `pc` bit at the end.

Judging from your other comment, it sounds like these are two distinct architectures that need to distinguished further.  I'll update the `--list` output accordingly.

**Comment by nwolfe** [comment]
@johnduarte @LuvCurves Changed to `powerpc` here: 4c9905c003100446efcbaf27d636a81a9d5bbf37

**Comment by johnduarte** [comment]
:+1: thanks @nwolfe

**Comment by waynr** [comment]
> 447 files changes
If you change all these files every time you make a change to the repo then they become kind of meaningless. They are only useful as test fixtures if they don't change with the code you are changing. It is impossible to review every single change or even a meaningful portion of them.

I absolutely know it's possible to make the change you're trying to make here without changing every single test fixture so :-1:

**Comment by nwolfe** [comment]
@waynr Good idea with only committing the Huawei fixtures.

Added them here: https://github.com/puppetlabs/beaker-hostgenerator/pull/46/commits/ba951da2dec7efcceb85a0231a5a96913b4cdd5a

**Comment by waynr** [comment]
:+1:

**Pull request merged**